main.js

```diff
@@ -133,6 +153,35 @@ define(function (require, exports, module) {
         };
     }
     
+    /**
+     * Apply the whitespace colors.
+     * This does not overwrite styles already defined by a theme.
+     */
+    function _applyColors() {
+        // Compile the CSS from the template
+        var userColors = _preferences.get("colors"),
+            template   = _.template(stylesTemplate),
+            css = template({
+                "empty": userColors.light.empty,
+                "leading": userColors.light.leading,
+                "trailing": userColors.light.trailing,
+                "whitespace": userColors.light.whitespace,
+                
+                "darkEmpty": userColors.dark.empty,
+                "darkLeading": userColors.dark.leading,
+                "darkTrailing": userColors.dark.trailing,
+                "darkWhitespace": userColors.dark.whitespace
+            });
+
+        // If the styles do not exist, apply to the DOM
+        // Otherwise, update the existing element
+        if (!_styleInline) {
+            _styleInline = ExtensionUtils.addEmbeddedStyleSheet(css);
+        } else {
+            $(_styleInline).text(css);
```

Comment:
Nice. I like you optimized for this. You could create the embedded style element during the initialization phase if you want to completely avoid the runtime check each time `_applyColors` is called.  No biggie though.